targets/TARGET_STM/lp_ticker.c

```diff
@@ -191,13 +198,15 @@ void lp_ticker_set_interrupt(timestamp_t timestamp)
     LptimHandle.Instance = LPTIM1;
     irq_handler = (void (*)(void))lp_ticker_irq_handler;
 
-    __HAL_LPTIM_CLEAR_FLAG(&LptimHandle, LPTIM_FLAG_CMPOK);
-    __HAL_LPTIM_COMPARE_SET(&LptimHandle, timestamp);
     /* CMPOK is set by hardware to inform application that the APB bus write operation to the LPTIM_CMP register has been successfully completed */
     /* Any successive write before the CMPOK flag be set, will lead to unpredictable results */
-    while (__HAL_LPTIM_GET_FLAG(&LptimHandle, LPTIM_FLAG_CMPOK) == RESET) {
+    if (__HAL_LPTIM_GET_FLAG(&LptimHandle, LPTIM_FLAG_CMPOK) == RESET) {
+        return;
```

Comment:
I wonder if this should actually be

    MBED_ASSERT(__HAL_LPTIM_GET_FLAG(&lptimHandle, LPTIM_FLAG_CMPOK));

How confident are you that it's working?

Reply:
> How confident are you that it's working?

I only executed all mbed-os tests

The other idea I have is to call lp_ticker_fire_interrupt if lp_ticker_set_interrupt is called too early (before LPTIM_FLAG_CMPOK) ?

Reply:
I'll leave exact behaviour up to @c1728p9, but it needs to do something other than just return silently in this case.

I would imagine assert should be okay as long as you have correctly configured `LPTICKER_DELAY_TICKS`. It will be a hard run-time check that your magic 3 is correct in practice.